docs/ktx/firestore.md

```diff
@@ -0,0 +1,67 @@
+# Firestore Kotlin Extensions
+
+## Getting Started
+
+To use the Cloud Firestore Android SDK with Kotlin Extenstions, add the following
+to your app's `build.gradle` file:
+
+```groovy
+// This library transitively includes the Firestore Android SDK
+implementation 'com.google.firebase:firebase-firestore-ktx:18.1.0
+```
+
+## Features
+
+### Convert a DocumentSnapshot field to a POJO
+
+**Kotlin**
+```kotlin
+val snapshot: DocumentSnapshot = ...
+val myObject = snapshot.get("fieldPath", MyClass::class.java)
+```
+
+**Kotlin + KTX**
+```kotlin
+val snapshot: DocumentSnapshot = ...
+val myObject = snapshot.get<MyClass>("fieldPath")
+```
+
+### Get an instance of FirebaseFirestore
+
+**Kotlin**
+```kotlin
+val firestore = FirebaseFirestore.getInstance()
+```
+
+**Kotlin + KTX**
+```kotlin
+val firestore = Firebase.firestore
```

Comment:
Might be a good idea to add an overload for custom app?

e.g:

**Kotlin**
```kotlin
val firestore = FirebaseFirestore.getInstance()
val anotherFirestore = FirebaseFirestore.getInstance(FirebaseApp.getInstance("myApp"))
```

**Kotlin + KTX**
```kotlin
val firestore = Firebase.firestore
val anotherFirestore = Firebase.firestore(Firebase.app("myApp"))